build/webpack/plugins/progress-plugin/baseProgressIndicator.js

```diff
@@ -0,0 +1,105 @@
+'use strict';
+
+/*!
+ * V4Fire Client Core
+ * https://github.com/V4Fire/Client
+ *
+ * Released under the MIT license
+ * https://github.com/V4Fire/Client/blob/master/LICENSE
+ */
+
+const chalk = require('chalk');
+
+/**
+ * Class for simple logging webpack build progress
+ * print status in console at every change of progress
+ */
+class BaseProgressIndicator {
+	constructor() {
+		this.handlers = {};
+		this.finishedHandlersCounter = 0;
+		this.startTime = new Date();
+		this.enableLog = true;
+	}
+
+	/**
+	 * Wrapper of updating method for webpack
+	 *
+	 * @param {string} name
+	 * @param {number} value
+	 */
+	webpackHandler(name, value) {
+		this.updateProgress(name, value);
+	}
+
+	/**
+	 * Convert value into percentage representation
+	 *
+	 * @param {number} value
+	 */
+	convertToPercentage(value) {
+		return Math.floor(value * 100);
+	}
+
+	/**
+	 * Add handler for logging build process
+	 *
+	 * @param {string} name
+	 */
+	addHandler(name) {
+		this.handlers[name] = true;
+
+		if (this.enableLog) {
+			console.log(`Started build step ${name}`);
+		}
+
+		return this.webpackHandler.bind(this, name);
+	}
+
+	/**
+	 * Updates status of build by name
```

Comment:
Нельзя забывать про артикли, т.к. они подсвечивают объекты над которыми идет работа.

```suggestion
	 * Updates a build progress status by the passed name
```